### PR TITLE
config: do not allow negative validators/committee count

### LINKED ISSUE #2946: Node starts with bad config and panics
I tried to run node with next config:
```yaml
ProtocolConfiguration:
  ValidatorsHistory:
    0: -1
```

```
$ neo-go node --config-path ./
```

The application started and almost immediately threw panic:
```
panic: runtime error: slice bounds out of range [:-1]

goroutine 1 [running]:
github.com/nspcc-dev/neo-go/pkg/core/native.(*NEO).updateCache(0xc0004102c0, 0xc0001330e0, {0xc000118140?, 0x1?, 0xa49d65?}, 0x6)
	github.com/nspcc-dev/neo-go/pkg/core/native/native_neo.go:349 +0x207
github.com/nspcc-dev/neo-go/pkg/core/native.(*NEO).InitializeCache(0xc0004102c0, 0x134370?, 0xc000134f50?)
	github.com/nspcc-dev/neo-go/pkg/core/native/native_neo.go:320 +0x129
github.com/nspcc-dev/neo-go/pkg/core.(*Blockchain).initializeNativeCache(0xc00016ac00, 0x6?, 0x1?)
	github.com/nspcc-dev/neo-go/pkg/core/blockchain.go:885 +0x31
github.com/nspcc-dev/neo-go/pkg/core.(*Blockchain).init(0xc00016ac00)
	github.com/nspcc-dev/neo-go/pkg/core/blockchain.go:467 +0x648
github.com/nspcc-dev/neo-go/pkg/core.NewBlockchain({0x1268cc0?, 0xc000130058}, {{0x0, 0x0, 0xddb1, 0xc350, 0xc0001232c0, 0x12795f58d50000, 0x0, 0x0, ...}, ...}, ...)
	github.com/nspcc-dev/neo-go/pkg/core/blockchain.go:323 +0x191b
github.com/nspcc-dev/neo-go/cli/server.initBlockChain({{0x0, 0x0, 0xddb1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...}, ...)
	github.com/nspcc-dev/neo-go/cli/server/server.go:661 +0x245
github.com/nspcc-dev/neo-go/cli/server.initBCWithMetrics({{0x0, 0x0, 0xddb1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...}, ...}, ...)
	github.com/nspcc-dev/neo-go/cli/server/server.go:135 +0x57
github.com/nspcc-dev/neo-go/cli/server.startServer(0xc000400160)
	github.com/nspcc-dev/neo-go/cli/server/server.go:454 +0x53c
github.com/urfave/cli.HandleAction({0xe6c440?, 0x1129500?}, 0x4?)
	github.com/urfave/cli@v1.22.5/app.go:524 +0x50
github.com/urfave/cli.Command.Run({{0xff4fe7, 0x4}, {0x0, 0x0}, {0x0, 0x0, 0x0}, {0x1003f24, 0x12}, {0x102a7c6, ...}, ...}, ...)
	github.com/urfave/cli@v1.22.5/command.go:173 +0x65b
github.com/urfave/cli.(*App).Run(0xc000360000, {0xc00003e0c0, 0x4, 0x4})
	github.com/urfave/cli@v1.22.5/app.go:277 +0x8a7
main.main()
	./main.go:12 +0x33
```

I understand that configuration is incorrect, but value type is `map[uint32]int` according to [`docs`](https://github.com/nspcc-dev/neo-go/blob/v0.101.0/docs/node-configuration.md#protocol-configuration), so technically negative values are possible.

IMO the best way is to change type to `map[uint32]uint32` which will be compatible with current working configs. We can also add check and reaction.

Maybe `CommitteeHistory` with the same type has similar problem, I didn't check. Maybe there is more.

```
$ neo-go --version
NeoGo
Version: 0.101.0
GoVersion: go1.19.
```

### PR DESCRIPTION
**Pull request by AnnaShaleva**
Close #2946.

### CODE REVIEW

File: docs/node-configuration.md [hunk 1]
```diff
@@ -365,7 +365,7 @@ protocol-related settings described in the table below.
 | StateRootInHeader | `bool` | `false` | Enables storing state root in block header. | Experimental protocol extension! |
 | StateSyncInterval | `int` | `40000` | The number of blocks between state heights available for MPT state data synchronization. | `P2PStateExchangeExtensions` should be enabled to use this setting. |
 | TimePerBlock | `Duration` | `15s` | Minimal (and targeted for) time interval between blocks. Must be an integer number of milliseconds. |
-| ValidatorsCount | `int` | `0` | Number of validators set for the whole network lifetime, can't be set if `ValidatorsHistory` setting is used. |
+| ValidatorsCount | `uint32` | `0` | Non-negative number of validators set for the whole network lifetime, can't be set if `ValidatorsHistory` setting is used. |
```
Comment (roman-khimov) [inline docs/node-configuration.md:368]: Ditto.

File: docs/node-configuration.md [hunk 2]
```diff
@@ -341,7 +341,7 @@ protocol-related settings described in the table below.
...
-| CommitteeHistory | map[uint32]int | none | Number of committee members after the given height, for example `{0: 1, 20: 4}` sets up a chain with one committee member since the genesis and then changes the setting to 4 committee members at the height of 20. `StandbyCommittee` committee setting must have the number of keys equal or exceeding the highest value in this option. Blocks numbers where the change happens must be divisible by the old and by the new values simultaneously. If not set, committee size is derived from the `StandbyCommittee` setting and never changes. |
+| CommitteeHistory | map[uint32]uint32 | none | Positive number of committee members after the given height, for example `{0: 1, 20: 4}` sets up a chain with one committee member since the genesis and then changes the setting to 4 committee members at the height of 20. `StandbyCommittee` committee setting must have the number of keys equal or exceeding the highest value in this option. Blocks numbers where the change happens must be divisible by the old and by the new values simultaneously. If not set, committee size is derived from the `StandbyCommittee` setting and never changes. |
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply docs/node-configuration.md]: This was added to denote that zero number is not allowed as far.

Fixed.

File: docs/node-configuration.md [hunk 3]
```diff
@@ -366,6 +366,6 @@ protocol-related settings described in the table below.
 | StateSyncInterval | `int` | `40000` | The number of blocks between state heights available for MPT state data synchronization. | `P2PStateExchangeExtensions` should be enabled to use this setting. |
 | TimePerBlock | `Duration` | `15s` | Minimal (and targeted for) time interval between blocks. Must be an integer number of milliseconds. |
 | ValidatorsCount | `int` | `0` | Number of validators set for the whole network lifetime, can't be set if `ValidatorsHistory` setting is used. |
-| ValidatorsHistory | map[uint32]int | none | Number of consensus nodes to use after given height (see `CommitteeHistory` also). Heights where the change occurs must be divisible by the number of committee members at that height. Can't be used with `ValidatorsCount` not equal to zero. |
+| ValidatorsHistory | map[uint32]uint32 | none | Positive number of consensus nodes to use after given height (see `CommitteeHistory` also). Heights where the change occurs must be divisible by the number of committee members at that height. Can't be used with `ValidatorsCount` not equal to zero. |
```
[the remark this answers is not in the document]
Comment (AnnaShaleva) [reply docs/node-configuration.md]: Fixed.

File: docs/node-configuration.md (hunk 1 above)
Comment (AnnaShaleva) [reply docs/node-configuration.md]: Fixed.

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**